python/plugins/db_manager/db_plugins/html_elems.py

```diff
@@ -36,7 +36,12 @@ def toHtml(self):
         if hasattr(self.data, 'toHtml'):
             return self.data.toHtml()
 
-        html = unicode(self.data).replace("\n", "<br>")
+        if isinstance(self.data, str):
+            html = unicode(self.data, encoding='utf-8', errors='replace')
+        else:
+            html = unicode(self.data)
```

Comment:
In this case it already is a unicode object, right? Just do

html = self.data

Reply:
Right, besides str and int it's sometimes unicode (and maybe other types I didn't observe). I've added an explicit case for unicode to make you happy ;-p although I'd prefer to touch the 2.18 code as little as possible, especially that it's not covered by tests and I'm not sure what datatypes may happen here.

Reply:
I fully agree with you, and even more - as also those simplifications introduced new problems ;) At least when switching from unicode to str in Processing some months ago... 
The function I use the most often is probably:
``nunicode = lambda s: unicode(s) if s is not None else None`` 
;)

Reply:
That doesn't look like something you should be using ;)

Merge away!